.github/workflows/ci.yml

```
@@ -350,3 +350,41 @@ jobs:
           CIBW_BUILD_VERBOSITY: 1
         with:
           package-dir: examples/namespace_package
+
+  test-mingw:
+    runs-on: windows-latest
+    name:  ${{ matrix.python-version }} mingw-${{ matrix.arch }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include: [
+          { msystem: MINGW64, arch: x86_64, path: mingw64, rust_target: x86_64-pc-windows-gnu },
+          { msystem: MINGW32, arch: i686, path: mingw32, rust_target: i686-pc-windows-gnu }
+        ]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install MSys2 and dependencies
+        uses: msys2/setup-msys2@v2
+        with:
+          update: true
+          msystem: ${{ matrix.msystem }}
+          install: >-
+            mingw-w64-${{ matrix.arch }}-python
+            mingw-w64-${{ matrix.arch }}-python-pip
+            mingw-w64-${{ matrix.arch }}-openssl
+            mingw-w64-${{ matrix.arch }}-toolchain
+
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          target: ${{ matrix.rust-target }}
+
+      - name: Install test dependencies
+        run: pip install --upgrade nox
+
+      - name: Test examples
+        shell: bash
```

I think you need

```suggestion
        shell: msys2 {0}
```

See https://github.com/msys2/setup-msys2#usage

---

Thanks!